resources/custom-resources/keycloak-setup/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/KeycloakSetup.java

```diff
@@ -0,0 +1,223 @@
+    protected boolean importRealm(String keycloakHost, String bearerToken, String realmName,
+                                              String username, String password, String email) {
+        // https://www.keycloak.org/docs-api/19.0.3/rest-api/index.html#_realms_admin_resource
+
+        LinkedHashMap<String, Object> adminUser = new LinkedHashMap<>();
+        adminUser.put("enabled", "true");
+        adminUser.put("createdTimestamp", LocalDateTime.now().toEpochSecond(ZoneOffset.UTC));
+        adminUser.put("username", username);
+        adminUser.put("email", email);
+        adminUser.put("emailVerified", "true");
+        adminUser.put("credentials", List.of(
+                Map.of("type", "password", "temporary", "true", "value", password)
+        ));
+        adminUser.put("requiredActions", List.of("UPDATE_PASSWORD"));
+
+        LinkedHashMap<String, Object> realm = new LinkedHashMap<>();
+        realm.put("realm", realmName);
+        realm.put("enabled", "true");
+        realm.put("users", List.of(adminUser));
+
+        try {
+            URI keycloakImportRealmEndpoint = new URI(keycloakHost + "/admin/realms");
+            String body = Utils.toJson(realm);
+            HttpRequest request = HttpRequest.newBuilder()
+                    .version(HttpClient.Version.HTTP_1_1) // EOF reached while reading due to chunked transfer-encoding
+                    .uri(keycloakImportRealmEndpoint)
+                    .setHeader("Authorization", "Bearer " + bearerToken)
+                    .setHeader("Content-Type", "application/json")
+                    .POST(HttpRequest.BodyPublishers.ofString(body))
+                    .build();
+
+            LOGGER.info("Invoking Keycloak endpoint {}", request.uri());
+            HttpClient client = HttpClient.newBuilder().build();
+            HttpResponse<String> response = client.send(request,
+                    HttpResponse.BodyHandlers.ofString(StandardCharsets.UTF_8));
+            if (HttpURLConnection.HTTP_CREATED == response.statusCode()) {
+                return true;
+            } else {
+                LOGGER.error("Received HTTP status " + response.statusCode());
+                LOGGER.error(response.body());
+                throw new RuntimeException("Keycloak import realm failed HTTP " + response.statusCode());
+            }
+        } catch (URISyntaxException | IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected Map<String, Object> adminPasswordGrant(String keycloakHost, String username, String password) {
+        Map<String, Object> passwordGrant = null;
+        try {
+            URI keycloakAdminTokenEndpoint = new URI(keycloakHost
+                    + "/realms/master/protocol/openid-connect/token");
+            String body = "grant_type=password"
+                    + "&client_id=admin-cli"
+                    + "&username=" + username
+                    + "&password=" + password;
```

Comment:
until we're using HTTPS this means we're sending password in plaintext over the wire, correct?

Reply:
That's correct - this is a not safe for production intermediate check in. We can certainly leave this PR open until all outstanding issues are resolved.